.github/workflows/ci.yml

```diff
@@ -0,0 +1,98 @@
+name: C++ CI Workflow
+
+on:
+  push:
+  pull_request:
+  schedule:
+  # * is a special character in YAML so you have to quote this string
+  # Execute a "nightly" build at 2 AM UTC 
+  - cron:  '0 2 * * *'
+  
+  
+jobs:
+  build:
+    name: '[${{ matrix.os }}@${{ matrix.build_type }}@yarp:${{ matrix.yarp_version }}@gazebo:${{ matrix.gazebo_version }}]'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        build_type: [Release, Debug]
+        os: [ubuntu-latest, macOS-latest]
+        gazebo_version: [gazebo9, gazebo10]
+        yarp_version: [devel]
+
+    steps:
+    - uses: actions/checkout@master
+        
+    # Print environment variables to simplify development and debugging
+    - name: Environment Variables
+      shell: bash
+      run: env
+        
+    # ============
+    # DEPENDENCIES
+    # ============
+            
+    - name: Dependencies [macOS]
+      if: matrix.os == 'macOS-latest'
+      shell: bash
+      run: |
+        brew cask install xquartz
+        brew install ace eigen opencv@3 osrf/simulation/${{ matrix.gazebo_version }}
+    
+    - name: Dependencies [Ubuntu]
+      if: matrix.os == 'ubuntu-latest'
+      shell: bash
+      run: |
+        sudo sh -c 'echo "deb http://packages.osrfoundation.org/gazebo/ubuntu `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-latest.list'
+        wget http://packages.osrfoundation.org/gazebo.key -O - | sudo apt-key add -
+        sudo apt-get update
+        sudo apt-get install git build-essential cmake libace-dev libeigen3-dev libopencv-dev 
+        sudo apt-get install lib${{ matrix.gazebo_version }}-dev
+        
+    - name: Source-based Dependencies [Ubuntu/macOS] 
+      if: matrix.os == 'ubuntu-latest' || matrix.os == 'macOS-latest'
+      shell: bash
+      run: |
+        # YCM
+        git clone https://github.com/robotology/ycm
+        cd ycm
+        git checkout devel
+        mkdir -p build
+        cd build
+        cmake -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
+        cmake --build . --config ${{ matrix.build_type }} --target install 
+        # YARP
+        git clone https://github.com/robotology/yarp
```

Comment:
I would add `cd ${GITHUB_WORKSPACE}` here

Reply:
Good catch, fixed it in https://github.com/robotology/gazebo-yarp-plugins/pull/453/commits/3ec9d8b013f19de032d2049c371f05172126011b .